raiden-ts/src/path/epics.ts

```diff
@@ -127,55 +118,61 @@ const makeAndSignLastIOURequest$ = (sender: Address, receiver: Address, signer:
   });
 
 const prepareNextIOU$ = (
-  state: RaidenState,
   pfs: PFS,
-  deps: RaidenEpicDeps,
-  httpTimeout: number,
   tokenNetwork: Address,
+  state$: Observable<RaidenState>,
+  deps: RaidenEpicDeps,
 ): Observable<Signed<IOU>> => {
-  const cachedIOU: IOU | undefined = get(state.path, ['iou', tokenNetwork, pfs.address]);
-  return (cachedIOU
-    ? of(cachedIOU)
-    : makeAndSignLastIOURequest$(state.address, pfs.address, deps.signer).pipe(
-        mergeMap(payload =>
-          fromFetch(
-            `${pfs.url}/api/v1/${tokenNetwork}/payment/iou?${new URLSearchParams(
-              payload,
-            ).toString()}`,
-            {
-              method: 'GET',
-              headers: { 'Content-Type': 'application/json' },
-            },
-          ),
-        ),
-        timeout(httpTimeout),
-        mergeMap(async response => {
-          const text = await response.text();
-          if (response.status === 404) {
-            return makeIOU(
-              state.address,
-              pfs.address,
-              pfs.price,
-              bigNumberify(state.chainId) as UInt<32>,
-              await oneToNAddress(deps.userDepositContract),
-              state.blockNumber,
-            );
-          }
-          if (!response.ok)
-            throw new Error(`PFS: last IOU request: code=${response.status} => body="${text}"`);
+  return state$.pipe(
+    withLatestFrom(deps.config$),
+    first(),
+    switchMap(([state, { httpTimeout }]) => {
+      const cachedIOU: IOU | undefined = get(state.path.iou, [tokenNetwork, pfs.address]);
+      return (cachedIOU
+        ? of(cachedIOU)
+        : makeAndSignLastIOURequest$(deps.address, pfs.address, deps.signer).pipe(
+            mergeMap(payload =>
+              fromFetch(
+                `${pfs.url}/api/v1/${tokenNetwork}/payment/iou?${new URLSearchParams(
+                  payload,
+                ).toString()}`,
+                {
+                  method: 'GET',
+                  headers: { 'Content-Type': 'application/json' },
+                },
+              ).pipe(timeout(httpTimeout)),
```

Comment:
Thank you for taking the time to fix the timeout issue.

Reply:
I've hit it as well, so made sense.